app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb

```diff
@@ -0,0 +1,81 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # This class is responsible for fetching JWK Set from provider-uri
+      class FetchProviderUriSigningKey < FetchSigningKeyInterface
+
+        def initialize(
+            authentication_parameters:,
+            fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+            resource_class: ::Resource,
+            discover_identity_provider: Authentication::OAuth::DiscoverIdentityProvider.new,
+            logger: Rails.logger
+          )
+          @logger = logger
+          @fetch_required_secrets = fetch_required_secrets
+          @resource_class = resource_class
+          @discover_identity_provider = discover_identity_provider
+
+          @authentication_parameters = authentication_parameters
+        end
+
+        def valid_configuration?
+          return @valid_configuration if defined?(@valid_configuration)
+
+          @valid_configuration = provider_uri_resource_exists?
+        end
+
+        def call
+          discover_provider
+          fetch_provider_keys
+        end
+
+        private
```

Comment:
please re-order the methods according to their calls

Reply:
Done